File: StriverA2Z/C++/2_SortingTechniques/selection_sort.cpp
// O(n^2) - best, worst, avg

#include<bits/stdc++.h>

using namespace std;

void selectionSort(int arr[], int length) {
    for (int i = 0; i <= length - 2; i++) {
        int min = i;
        for (int j = i; j <= length - 1; j++) {
            if (arr[j] < arr[min]) {
                min = j;
            }
        }
        int temp = arr[min];
        arr[min] = arr[i];
        arr[i] = temp;
    }
}

int main() {
    int n;
    cout << "Enter the size of array: ";
    cin >> n;

    int arr[n];
    for (int i = 0; i < n; i++) {
        cout << "Enter the " << i << " position value:";
        cin >> arr[i];
    }

    selectionSort(arr, n);

    for (int x: arr) {
        cout << x << " ";
    }
    return 0;
}

File: StriverA2Z/Java/SortingAlgorithms/SelectionSort.java
// O(n^2)

public class SelectionSort {
    public static void main(String[] args) {
        int[] arr = {13, 46, 24, 52, 20, 9};
        selectionSort(arr);
    }

    static void selectionSort(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            int min = i;
            for (int j = i; j < arr.length; j++) {
                if (arr[min] > arr[j]) {
                    min = j;
                }
            }
            int temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
        }

        // for printing
        for (int x : arr) {
            System.out.print(x + " ");
        }
    }
}